app/services/RestaurantService.java

```diff
@@ -176,82 +227,122 @@ public Restaurant getRestaurantWithId(final UUID id) {
+		.add(Restrictions.eq("id", imageUploadForm.getRestaurantId())).uniqueResult();
+
+	String newImagePath = AWS_BASE_PATH + imageUploadForm.getRestaurantId() + "-" + imageUploadForm.getImageType()
+		+ "." + imageUploadForm.getExtension();
+
+	if (imageUploadForm.getImageType().equals("profile"))
+	{
+	    restaurant.setProfileImagePath(newImagePath);
+	} else
+	{
+	    restaurant.setCoverImagePath(newImagePath);
 	}
 
-	/**
-	 * Gets number of restaurants.
-	 *
-	 * @return the number of restaurants
-	 */
-	public Long getNumberOfRestaurants() {
-		return Long.valueOf(getSession().createCriteria(Restaurant.class)
-				.setProjection(Projections.rowCount())
-				.uniqueResult().toString());
-	}
+	getSession().update(restaurant);
+	return "{ \"imageFor\": \"" + imageUploadForm.getImageType() + "\", \"url\": \"" + newImagePath + "\"}";
+    }
 
-	/**
-	 * Update picture string.
-	 *
-	 * @param imageUploadForm the image upload form
-	 * @return the string
-	 * @throws Exception the exception
-	 */
-	public String updatePicture(final ImageUploadForm imageUploadForm) throws Exception {
-		Restaurant restaurant = (Restaurant) getSession().createCriteria(Restaurant.class)
-				.add(Restrictions.eq("id", imageUploadForm.getRestaurantId()))
-				.uniqueResult();
-
-		String newImagePath = AWS_BASE_PATH + imageUploadForm.getRestaurantId() + "-" + imageUploadForm.getImageType() + "." + imageUploadForm.getExtension();
-
-		if (imageUploadForm.getImageType().equals("profile")) {
-			restaurant.setProfileImagePath(newImagePath);
-		} else {
-			restaurant.setCoverImagePath(newImagePath);
-		}
+    @SuppressWarnings("unchecked")
+    private UserHabitsBean getUserHabits(UUID userId)
+    {
+
+	Integer price = 0;
+
+	UUID cityID = (UUID) getSession().createCriteria(User.class).setProjection(Projections.property("city.id"))
+		.uniqueResult();
+
+	List<RestaurantReview> restaurantReviews = getSession().createCriteria(RestaurantReview.class)
+		.add(Restrictions.eq("userId", userId)).list();
+
+	List<UUID> userReviewRestaurantIds = restaurantReviews.stream().map(RestaurantReview::getRestaurantId)
+		.collect(Collectors.toList());
+
+	List<Restaurant> userVisitedRestaurant = getSession().createCriteria(Restaurant.class)
+		.add(Restrictions.in("id", userReviewRestaurantIds)).list();
 
-		getSession().update(restaurant);
-		return "{ \"imageFor\": \"" + imageUploadForm.getImageType() + "\", \"url\": \"" + newImagePath + "\"}";
+	for (Restaurant restaurant : userVisitedRestaurant)
+	{
+	    price += restaurant.getPriceRange();
 	}
+	price /= userVisitedRestaurant.size();
+
+	UserHabitsBean userHabitsBean = (UserHabitsBean) getSession().createCriteria(RestaurantReview.class, "review")
+		.add(Restrictions.eq("userId", userId))
+		.setProjection(Projections.projectionList().add(Projections.groupProperty("review.userId").as("userId"))
+			.add(Projections.avg("review.rating").as("userRatingRange")))
```

Comment:
You could've just as easily calculated the average restaurant price range here. The RestaurantReview model has a reference to the Restaurant object. You only need an alias to access all other properties (apart from id, which is always visible)